node/core/pvf/src/execute/worker.rs

```diff
@@ -261,19 +266,34 @@ impl Response {
 
 /// The entrypoint that the spawned execute worker should start with. The `socket_path` specifies
 /// the path to the socket used to communicate with the host.
-pub fn worker_entrypoint(socket_path: &str) {
+pub fn worker_entrypoint(socket_path: &str, node_version: Option<&str>) {
 	worker_event_loop("execute", socket_path, |rt_handle, mut stream| async move {
-		let handshake = recv_handshake(&mut stream).await?;
+		let worker_pid = std::process::id();
+		let version_mismatch = if let Some(version) = node_version {
+			version != env!("SUBSTRATE_CLI_IMPL_VERSION")
+		} else {
+			false
+		};
 
+		let handshake = recv_handshake(&mut stream).await?;
 		let executor = Arc::new(Executor::new(handshake.executor_params).map_err(|e| {
 			io::Error::new(io::ErrorKind::Other, format!("cannot create executor: {}", e))
 		})?);
 
 		loop {
 			let (artifact_path, params, execution_timeout) = recv_request(&mut stream).await?;
+			if version_mismatch {
```

Comment:
Should document why we check this in the `loop` instead of where `version_mismatch` is set. It's the correct behavior but it's not immediately obvious why.

Reply:
Think we need to do another version check, between writing the temporary artifact and sending a success result back to the node.

Reply:
Should we add another version check on the node side, when receiving the temporary artifact, that it was built with the current node version, before renaming the artifact?

Reply:
> Should we add another version check on the node side

I don't think so, the new node instance wouldn't be able to communicate with an old worker as they communicate through unix socket created for every worker from scratch.